.circleci/config.yml

```diff
@@ -0,0 +1,123 @@
+version: 2.1
+
+orbs:
+  node: circleci/node@5.1.0
+  win: circleci/windows@5.0.0
+
+commands:
+  install:
+    steps:
+      - run: git config --global core.autocrlf input
+      - node/install:
+          node-version: '18.4.0'
+      - run: nvm use 18.4.0
+      - checkout
+      - restore_cache:
+          keys:
+            - v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+            - v1-dependencies-{{ arch }}
+      - run: npm install --global yarn
+      - run: yarn install --frozen-lockfile
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ arch }}-{{ checksum "yarn.lock" }}
+
+jobs:
+  lint:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - install
+      - run: yarn run lint
+  mac-build:
+    parameters:
+      arch:
+        type: enum
+        enum: [ 'x64', 'arm64' ]
+    macos:
+      xcode: '13.4.0'
+    resource_class: macos.x86.medium.gen2
+    steps:
+      - install
+      - run: yarn test:ci
+      - run: chmod +x tools/add-macos-cert.sh && ./tools/add-macos-cert.sh
+      - run: npx yarn make --arch=<< parameters.arch >>
+      - store_artifacts:
+          path: out
+      - persist_to_workspace:
+          root: '*'
+          paths:
+            - out
+  win-build:
+    parameters:
+      arch:
+        type: enum
+        enum: [ 'x64', 'arm64', 'ia32' ]
+    executor:
+      name: win/default
+      shell: bash.exe
+    steps:
+      - install
+      - run: yarn test:ci
+      - run: export WINDOWS_CODESIGN_FILE=$(node add-windows-cert.js)
+      - run: npx yarn make --arch=<< parameters.arch >>
+      - store_artifacts:
+          path: out
+      - persist_to_workspace:
+          root: '*'
+          paths:
+            - out
+  linux-build:
+    parameters:
+      arch:
+        type: enum
+        enum: [ 'x64', 'arm64','armv7l' ]
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run: sudo apt-get update && sudo apt install rpm
+      - install
+      - run: yarn test:ci
+      - run: npx yarn make --arch=<< parameters.arch >>
+      - store_artifacts:
+          path: out
+      - persist_to_workspace:
+          root: '*'
+          paths:
+            - out
+  maybe-publish-to-github:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - run: sudo apt-get update && sudo apt install rpm
+      - install
+      - run: yarn run publish
+
+workflows:
+  build-and-test:
+    jobs:
+      - lint
+      - mac-build:
+          matrix:
+            parameters:
+              arch: [ x64, arm64 ]
+      - win-build:
+          matrix:
+            parameters:
+              arch: [ x64, ia32 ]
+      - linux-build:
+          matrix:
+            parameters:
+              arch: [ x64, arm64, armv7l ]
+      - maybe-publish-to-github:
+          requires:
+            - mac-build
+            - win-build
+            - linux-build
```

Comment:
Let's also require `lint` for publish.

I think these have to be the full name with the arch, e.g. `linux-build-x64`, but I'm not positive. If the base name works, that's great, and would simplify other configs, but on other repos we require the full expanded matrix.